### ISSUE TITLE
Monotonicity enforcement

### ISSUE THREAD

**Issue by berland** [equinor/pyscal#158]
There are monotonicity requirements (non-strict, i.e. constant parts are allowed) from Eclipse for most columns in tables outputted by pyscal. 

Pyscal does not enforce that numerical corner cases can produce non-monotone results. Pyscal tests verify that the dataframes produced through test code yield dataframes that are monotone. However, since constant sections in the vectors are allowed, and representation errors are a real thing for floats on computers, there is some numerical slack on what is a constant section.

On top of this, what is given to Eclipse is not the dataframes directly, it is given through an ASCII representation produced through Pandas' `to_csv()` where `float_format="1.7f"`.

A monotonicity enforcement must therefore be on the ASCII representation. 

For performance reason, testing the ASCII representation on every output call is discouraged. 

If monotonicity errors occur after merging of #157, a possible solution could be to write a code that detects constant sections in dataframe columns, and ensuring they have the same floating point representation. This should have acceptable performance through filtering on `diff()` close to zero, setting almost equal values to `nan` and then forward_fill/backwards_fill in the dataframes.

**Comment by berland** [comment]
This problem is now gone for capillary pressure, fixed in #160

**Comment by berland** [comment]
Monotonicity error has appeared in the Fossekall dataset, for which a freak realization gives.
```
 @--  ERROR  AT TIME        0.0   DAYS    ( 1-MAR-2013):
 @           MONOTONICITY ERROR IN SWOF TABLE NUMBER   2                     
 @           OIL REL. PERM. VALUES SHOULD BE STRICTLY                        
 @           MONOTONIC DECREASING WHEN HYSTERESIS IS USED.  
```
A portion of the SWOF output is:
``` 
-- SCAL recommendation interpolation to 0.986096
-- pyscal: 0.6.1
-- SW     KRW       KROW      PC
0.1800000 0.0000000 1.0000000 5.0000000
0.1900000 0.0000001 0.9921023 3.6620831
.....
0.8700000 0.7740661 0.0000005 0.1462313
0.8800000 0.7887943 0.0000003 0.1432576
0.8900000 0.8057981 0.0000001 0.1402952
0.9000000 0.8235410 0.0000001 0.1377806   # crashes on this
0.9100000 0.8412778 0.0000000 0.1354983
0.9200000 0.8590062 0.0000000 0.1330228
```

**Comment by berland** [comment]
Pyscal has no knowledge of whether hysteresis is in use, so that means for pyscal to be able to fix this, it has to require oil relperm to be monotonic decreasing always.

**Comment by berland** [comment]
Solved in #160
